.github/pull_request_template.md

```diff
@@ -0,0 +1,10 @@
+Pull Request Checklist
+ - [ ] Pull request includes a description of the change and the reason behind it.
+ - [ ] Pull request [uses keywords to close relevant issues](https://help.github.com/en/articles/closing-issues-using-keywords).
+ - [ ] Pull request includes unit tests for any new functionality.
+ - [ ] `CHANGELOG.md` has been updated.
+ - [ ] `_version.py` has been updated.
```

Comment:
**TL;DR:**

For this one, you should point to some notes (either in a `CONTRIBUTING.md` or some standard on the internet like https://semver.org/) explaining how people should reason about version updates.

**Longer explanation**

One other thing to think about...typically, package maintainers will allow many changes to accumulate on GitHub and then treat it as the **sole responsibility of maintainers** to manually bump the version at release time.

So, for example, if the latest version of a project on PyPi is 0.1.7 then the version in the repo will be 0.1.7.9999 and at release time the maintainer will manually change it to 0.2.0, do a release, then make the version on master in the repo 0.2.0.9999.

Since you are not distributing this with a package manager, this probably doesn't make sense for you. But once we start publishing the app side to Dockerhub, you'll want to version it and there will be an artifact repository in play (that's conceptually similar to a package manager).

Reply:
> typically, package maintainers...treat it as the sole responsibility of maintainers to manually bump the version at release time.

That seems reasonable.

> So, for example, if the latest version of a project on PyPi is 0.1.7 then the version in the repo will be 0.1.7.9999

Why the `.9999`?

Reply:
That 4th component of the version has no semantic meaning, so there's no risk of someone interpreting it as "something has changed" in any meaningful way.

The .9999 is to reflect the idea that you aren't making any effort to differentiate between versions that are between formal semver version updates (basically, you're allowing changes to accumulate on `master` in source control until you do a release).

It just has to be "something greater than 0" so that when people install from source, if they want to, their local package management actually upgrades. So e.g. if you got `0.2.0` from PyPi and then cloned a repo where version was v0.2.0, some methods of installation would say "oh you already have 0.2.0, I'm not going to update it"